specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-07-01/managedClusters.json

```diff
@@ -7461,6 +7557,109 @@
+    },
+    "Machine": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
+        }
+      ],
+      "description": "A machine. Contains details about the underlying virtual machine. A machine may be visible here but not in kubectl get nodes; if so it may be because the machine has not been registered with the Kubernetes API Server yet.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/MachineProperties",
+          "description": "The properties of the machine"
+        }
+      },
+      "type": "object"
+    },
+    "MachineIpAddress": {
+      "description": "The machine IP address details.",
+      "properties": {
+        "family": {
+          "description": "To determine if address belongs IPv4 or IPv6 family",
+          "enum": [
+            "IPv4",
+            "IPv6"
+          ],
+          "readOnly": true,
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "IPFamily",
+            "values": [
+              {
+                "description": "IPv4 family",
+                "value": "IPv4"
+              },
+              {
+                "description": "IPv6 family",
+                "value": "IPv6"
+              }
+            ]
+          }
+        },
+        "ip": {
+          "description": "IPv4 or IPv6 address of the machine",
+          "readOnly": true,
+          "type": "string"
+        }
+      },
+      "readOnly": true,
+      "type": "object"
+    },
+    "MachineListResult": {
+      "description": "The response from the List Machines operation.",
+      "properties": {
+        "nextLink": {
+          "description": "The URL to get the next set of machine results.",
+          "format": "uri",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "The list of Machines in cluster.",
+          "items": {
+            "$ref": "#/definitions/Machine"
+          },
+          "type": "array",
+          "x-ms-identifiers": [
+            "id"
+          ]
+        }
+      },
+      "type": "object"
+    },
+    "MachineNetworkProperties": {
+      "description": "network properties of the machine",
+      "properties": {
+        "ipAddresses": {
+          "description": "IPv4, IPv6 addresses of the machine",
+          "items": {
+            "$ref": "#/definitions/MachineIpAddress"
+          },
+          "readOnly": true,
+          "type": "array",
+          "x-ms-identifiers": []
+        }
+      },
+      "readOnly": true,
+      "type": "object"
+    },
+    "MachineProperties": {
+      "description": "The properties of the machine",
+      "properties": {
+        "network": {
+          "$ref": "#/definitions/MachineNetworkProperties"
+        },
+        "resourceId": {
+          "description": "Arm resource id of the machine. It can be used to GET underlying VM Instance",
```

Comment:
minor: You should specify the `format` here as `"format": "arm-id",`, since this is an ARM ID.

Reply:
See other examples in the managedCLusters.json file that do this, for example `sourceResourceId` of TrustedAccessRoleBindings.